File: web/src/types/position.ts
// 岗位相关类型定义
// 所有 ID 字段使用 string 类型，避免 JavaScript Number 精度丢失问题

// 岗位信息
export interface Position {
  id: string // 岗位 ID
  name: string // 岗位名称
  code: string // 岗位编码
  sort?: number // 排序
  status?: number // 状态：1-启用 0-禁用
  description?: string // 描述
  createdAt?: string // 创建时间
  updatedAt?: string // 更新时间
}

// 岗位列表响应
export interface PositionListResponse {
  list: Position[]
  total: number
  page: number
  pageSize: number
}

// 创建岗位请求
export interface CreatePositionRequest {
  name: string
  code: string
  sort?: number
  status?: number
  description?: string
}

// 更新岗位请求
export interface UpdatePositionRequest {
  id: string
  name?: string
  code?: string
  sort?: number
  status?: number
  description?: string
}

File: server/internal/domain/base/entity/position.go
package entity

import "time"

// Position 岗位领域实体
// 实现组织架构中的岗位管理功能
// 纯业务模型，无 GORM 标签
type Position struct {
	ID          int64     // 岗位 ID
	Name        string    // 岗位名称
	Sort        int       // 排序
	Status      int       // 状态：1-正常，0-禁用
	Description string    // 岗位描述
	CreatedBy   int64     // 创建人 ID
	CreatedAt   time.Time // 创建时间
	UpdatedBy   int64     // 更新人 ID
	UpdatedAt   time.Time // 更新时间
}

// IsActive 检查岗位是否激活
func (p *Position) IsActive() bool {
	return p.Status == 1
}
